src/common/catalog-entities/kubernetes-cluster.ts

```diff
@@ -110,9 +111,17 @@ export class KubernetesCluster extends CatalogEntity<CatalogEntityMetadata, Kube
         {
           title: "Delete",
           icon: "delete",
-          onClick: () => {
-            requestMain(clusterDeleteHandler, this.metadata.uid);
-            context.hideDetails();
+          onClick: async () => {
+            const undo = HotbarStore.getInstance().removeAllHotbarItems(this.getId());
+            
+            try {
+              context.hideDetails();
+  
+              await requestMain(clusterDeleteHandler, this.metadata.uid);
+            } catch (err) {
+              undo();
+              throw err;
```

Comment:
maybe just log instead of the undo and throw?

Reply:
Sure